### PR TITLE
test(job-scheduler): add tests covering job retries and stalls

### PR DESCRIPTION
**Pull request by manast**

### CODE REVIEW

File: tests/test_job_scheduler.ts [hunk 1]
```diff
@@ -1403,6 +1403,247 @@ describe('Job Scheduler', function () {
...
+            skipStalledCheck: true,
+          },
+        );
+      });
+
+      await processing;
+
+      // force remove the lock
+      const client = await queue.client;
+      const lockKey = `bull:${queueName}:${repeatableJob!.id}:lock`;
```
Comment (roggervalf) [inline tests/test_job_scheduler.ts:1588]: ${prefix} is missing as dragonfly requires it for tests to pass

File: tests/test_job_scheduler.ts [hunk 2]
```diff
@@ -1403,6 +1403,247 @@ describe('Job Scheduler', function () {
...
+      await client.del(lockKey);
+
+      const stalledCheckerKey = `bull:${queueName}:stalled-check`;
```
Comment (roggervalf) [inline tests/test_job_scheduler.ts:1591]: same as above

File: tests/test_job_scheduler.ts [hunk 3]
```diff
@@ -1403,6 +1403,247 @@ describe('Job Scheduler', function () {
     });
   });
 
+  describe('when repeatable job fails', async function () {
```
Comment (fgozdz) [inline tests/test_job_scheduler.ts:1406]: describe should not be async: https://mochajs.org/#limitations-of-asynchronous-callbacks

**Review by fgozdz — CHANGES_REQUESTED** [review]

**Comment by manast** [comment]
Thanks for the comments, I addressed all of them.

File: tests/test_queue.ts [hunk 4]
```diff
@@ -37,6 +37,20 @@ describe('queues', function () {
     await connection.quit();
   });
 
+  describe('use generics', function () {
+    it('should be able to use generics', async function () {
+      const queue = new Queue<{ foo: string; bar: number }>(queueName, {
+        connection,
```
Comment (roggervalf) [inline tests/test_queue.ts:43]: prefix is missing to be passed

**Review by roggervalf — APPROVED** [review]

**Pull request merged**